scripts/apps/search/controllers/MultiActionBarController.js

```diff
@@ -130,4 +130,21 @@ export function MultiActionBarController(
     this.canHighlightItems = function() {
         return multi.getItems().every((item) => authoring.itemActions(item).mark_item_for_highlight);
     };
+
+    /**
+     * Publish all items
+     */
+    this.publish = () => {
+        $q.all(multi.getItems().map((item) => authoring.publish(item, item)))
+            .then((responses) => {
+                let errors = responses.filter((response) => response.status >= 400);
```

Comment:
You can use `some` instead of `filter`. It'll return a boolean and it will stop as soon as it finds the first one.